### DISCUSSION TITLE
Is there a way to get "timedif" just like "datedif"

### DISCUSSION THREAD

**Discussion by cuihee** [handsontable/hyperformula#1461] [API Questions]
https://hyperformula.handsontable.com/guide/built-in-functions.html#list-of-available-functions 
there is DATEDIF, but no timedif?

maybe like
=timedif(45616.0, 45616.3333, "H")
=8

**Comment by AMBudnik** [comment]
We do not have `timedif` among the list of available functions.

However, if you would like to create one for your own purpose, I recommend reading our custom functions guide at https://hyperformula.handsontable.com/guide/custom-functions.html#working-demo.

**Comment by cuihee** [comment]
yes, I create it , mainly copy from your DATEDIF
thank you for your efficiency reply

this is my mainly code
`private timeNumber2timedif(timeNumber: number): timediftype {
    const 小数部分 = timeNumber - Math.trunc(timeNumber);
    const totalSeconds = Math.trunc(小数部分 * 86400);
    const h = Math.trunc(totalSeconds / 3600);
    const m = Math.trunc((totalSeconds % 3600) / 60);
    const s = totalSeconds % 60;
    return { h, m, s };
  }
  timeDif(ast: any, state: any) {
    return this.runFunction(
      ast.args,
      state,
      this.metadata("TIMEDIF"),
      (startTime: number, endTime: number, unit: string) => {
        debugger
        if (startTime > endTime) {
          return new CellError(ErrorType.NUM, ErrorMessage.StartEndDate);
        }
        const start = this.timeNumber2timedif(startTime);
        const end = this.timeNumber2timedif(endTime);
        switch (unit) {
          case "H":
            if (end.m > start.m || (end.m === start.m && end.s >= start.s)) {
              return end.h - start.h;
            } else {
              return end.h - start.h - 1;
            }
            break;
          case "M":
            return (
              (end.h - start.h) * 60 +
              (end.m - start.m) -
              (end.s < start.s ? 1 : 0)
            );
            break;
          case "S":
            const end_type = this.timeNumber2timedif(endTime)
            const start_type = this.timeNumber2timedif(startTime)
            const endTimeTotalSeconds = end_type.h * 3600 + end_type.m * 60 + end_type.s
            const startTimeTotalSeconds = start_type.h * 3600 + start_type.m * 60 + start_type.s
            return endTimeTotalSeconds - startTimeTotalSeconds
            break;

          default:
            return new CellError(ErrorType.NUM, ErrorMessage.BadMode);
            break;
        }
      }
    );
  }`

**Reply by AMBudnik** [reply]
Thank you for sharing. This may help other users who have the same need.
